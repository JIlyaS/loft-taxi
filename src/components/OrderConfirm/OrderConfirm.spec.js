import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { OrderConfirm } from './';

describe('OrderConfirm', () => {

  it('renders correctly snapshot', () => {
    const tree = renderer.create(<OrderConfirm setOrderSuccessAction={jest.fn()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render correct', () => {
    const { queryByTestId } = render(<OrderConfirm setOrderSuccessAction={jest.fn()} />);
    expect(queryByTestId('order-confirm')).toBeInTheDocument();
    expect(queryByTestId('order-confirm-btn')).toBeInTheDocument();
  });

  it('Click new order taxi', () => {
    const { queryByTestId } = render(<OrderConfirm setOrderSuccessAction={jest.fn()} />);
    const orderConfirmBtn = queryByTestId('order-confirm-btn');
    expect(orderConfirmBtn).toBeInTheDocument();
    fireEvent.click(orderConfirmBtn);
    // expect(queryByTestId('map-form')).toBeInTheDocument();
  });
});