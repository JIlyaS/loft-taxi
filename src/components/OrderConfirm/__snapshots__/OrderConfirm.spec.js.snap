// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderConfirm renders correctly snapshot 1`] = `
<div
  className="order-confirm"
  data-testid="order-confirm"
>
  <h2
    className="order-confirm__header"
  >
    Заказ размещен
  </h2>
  <p
    className="order-confirm__text"
  >
    Ваше такси уже едет к вам. Прибудет приблизительно через 10 минут.
  </p>
  <div
    className="order-confirm__block-btn"
  >
    <button
      className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-1 MuiButton-contained order-confirm__btn MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
      data-testid="order-confirm-btn"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Сделать новый заказ
      </span>
    </button>
  </div>
</div>
`;
